src/setup-java.ts

```diff
@@ -1,63 +1,72 @@
 import * as core from '@actions/core';
-import * as installer from './installer';
 import * as auth from './auth';
 import * as gpg from './gpg';
 import * as constants from './constants';
 import * as path from 'path';
+import {JavaInitOptions} from './distributors/base-installer';
+import {getJavaDistributor} from './distributors/distributor-factory';
 
 async function run() {
   try {
-    let version = core.getInput(constants.INPUT_VERSION);
-    if (!version) {
-      version = core.getInput(constants.INPUT_JAVA_VERSION, {required: true});
-    }
-
+    const version = core.getInput(constants.INPUT_JAVA_VERSION, {
+      required: true
+    });
     const arch = core.getInput(constants.INPUT_ARCHITECTURE, {required: true});
-    if (!['x86', 'x64'].includes(arch)) {
-      throw new Error(`architecture "${arch}" is not in [x86 | x64]`);
-    }
-
+    const javaDistributor = core.getInput('distribution');
     const javaPackage = core.getInput(constants.INPUT_JAVA_PACKAGE, {
       required: true
     });
     const jdkFile = core.getInput(constants.INPUT_JDK_FILE, {required: false});
 
-    await installer.getJava(version, arch, jdkFile, javaPackage);
+    const initOptions: JavaInitOptions = {
```

Comment:
May be `JavaInstallerOptions` or `JavaInstallerInputs` as a name for interface?

Reply:
Done.